### DISCUSSION TITLE
Why points size seem to change differently while zooming in and out in two rendering style?

### DISCUSSION THREAD

**Discussion by theganeshpandey** [pyvista/pyvista#5525] [Q&A] [answered]
Hi,

Say I plot some random points in two styles, `points_gaussian` and `points` in the same plotter.

```
import numpy as np
import pyvista as pv

points = np.random.rand(1000,3)
points_2 = np.random.rand(1000,3)

p = pv.Plotter()
p.background_color = 'k'
act_1 = p.add_points(points ,render_points_as_spheres=True, style='points_gaussian', color='aqua')
act_2 = p.add_points(points_2 ,render_points_as_spheres=True, style='points', color='red')
p.show()
```

Then in the rendered scene, when I scroll the mouse wheel to zoom in,  the cyan spheres become bigger and bigger while the red ones become smaller. How can we make both types of spheres behave similarly?  

It may be a stupid question but I would greatly appreciate someone explaining/fixing it.

**Comment by banesullivan** [comment, marked as answer]
The key here is that `points` and `points_gaussian` are two totally different rendering styles. 

1. `points` without `render_points_as_spheres=True` renders the points as little squares on the screen that remain a constant pixel size. So as you zoom in/out the `point_size` remains constant though perceptually it looks like they get bigger as you zoom out. When `render_points_as_spheres=True`, they are still constant size, just spheres.
2. `points_gaussian` is a special technique that renders points as individual soft sprites. You have the option of displaying these as tight “spheres” using `render_points_as_spheres=True` which is default here (note the difference with this being off by default for `poinnts`), or disabling it to create softer points at the expense of render performance. `points_gaussian` creates the size of the sprites or spheres based on a radius and this radius can vary per-point if you want (ref #4366). The radius is defined in "physical space" as opposed to screen space.

**Reply by MatthewFlamm** [reply]
I would add to this answer that another option is to explicitly generate spheres at the points using `glyph`.  This also renders the radius in "physical space" like `point_gaussian`.
